src/base.cc

```diff
@@ -63,15 +63,7 @@ void assertOK(AstObject *rawPtr1, AstObject *rawPtr2) {
 }
 
 ConstArray2D arrayFromVector(std::vector<double> const &vec, int nAxes) {
-    if (vec.size() % nAxes != 0) {
-        std::ostringstream os;
-        os << "vec length = " << vec.size() << " not a multiple of nAxes = " << nAxes;
-        throw std::runtime_error(os.str());
-    }
-    int nPoints = vec.size() / nAxes;
-    Array2D::Index shape = ndarray::makeVector(nPoints, nAxes);
-    Array2D::Index strides = ndarray::makeVector(nAxes, 1);
-    return external(vec.data(), shape, strides);
+    return arrayFromVector(const_cast<std::vector<double> &>(vec), nAxes);
 }
```

Comment:
I thought about templating but decided against it because the only way I could think to do it required two template parameters, one for the return value, one for the argument. These need to be matched to each other to avoid accepting a const vector and returning a non-const array. The argument parameter is also rather messy. So it seemed to me that the result would be hard to use and easy to misuse.

If there was a way to handle it with a single template parameter I'd be a lot more comfortable with the idea.

Would it help if I explicitly cast the return type using `static_cast`? (`const_cast` will not work)